### PR TITLE
Search api issue#32

### PR DESCRIPTION
**Pull request by FeizNouri**

### CODE REVIEW

File: gateway/src/main/java/com/hurence/webapiservice/historian/HistorianFields.java [hunk 1]
```diff
@@ -32,6 +32,8 @@ private HistorianFields() {}
     public static String RESPONSE_CHUNKS = "chunks";
     public static String RESPONSE_METRICS = "metrics";
     public static String RESPONSE_TOTAL_FOUND = "total_hit";
+    public static String RESPONSE_TOTAL_NAMES_FOUND = "total_names";
+    public static String RESPONSE_TOTAL_METRICS_FOUND = "total_metrics";
```
Comment (MiniPlayer) [inline gateway/src/main/java/com/hurence/webapiservice/historian/HistorianFields.java]: I would prefer those names. If you accept this you will have to modify your response file in tests. NAME and METRICS means the same thing, so i think this is ambiguous.
```suggestion
    public static String RESPONSE_TOTAL_METRICS_RETURNED = "total_returned";
    public static String RESPONSE_TOTAL_METRICS_FOUND = "total";
```

File: gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java [hunk 2]
```diff
@@ -239,20 +238,33 @@ public HistorianService getMetricsName(JsonObject params, Handler<AsyncResult<Js
...
+                        group.getGroupValue();
+                        LOGGER.debug(group.getGroupValue());
+                        metrics.add(group.getGroupValue());
+                        }
+                    }
+                LOGGER.debug("Found " + response.getRequestUrl() + response + " result : " + metrics);
                 LOGGER.debug("mertics :: "+ metrics);
                 p.complete(new JsonObject()
-                        .put(RESPONSE_TOTAL_FOUND, solrDocuments.getNumFound())
+                        .put(RESPONSE_TOTAL_METRICS_FOUND, totalMetricsFound)
```
Comment (MiniPlayer) [inline gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java]: There is a problem here as discussed in chat. RESPONSE_TOTAL_METRICS_FOUND makes no sense here as it is the number of chunk with a name matching one of the returned name.

Comment (FeizNouri) [reply gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java]: i think this is an old version of code

File: gateway/src/main/java/com/hurence/webapiservice/historian/HistorianFields.java (hunk 1 above)
Comment (FeizNouri) [reply gateway/src/main/java/com/hurence/webapiservice/historian/HistorianFields.java]: yes your choice is better

File: gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java [hunk 3]
```diff
@@ -239,20 +238,33 @@ public HistorianService getMetricsName(JsonObject params, Handler<AsyncResult<Js
...
+                    totalNamesFound = gc.getValues().size();
+                    List<Group> groups = gc.getValues();
+                    for(Group group : groups) {
+                        group.getGroupValue();
+                        LOGGER.debug(group.getGroupValue());
+                        metrics.add(group.getGroupValue());
+                        }
+                    }
+                LOGGER.debug("Found " + response.getRequestUrl() + response + " result : " + metrics);
                 LOGGER.debug("mertics :: "+ metrics);
```
Comment (lhubert) [inline gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java]: there is a typo in metrics ;) missing the r ;)

Comment (MiniPlayer) [inline gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java]:
```suggestion
                LOGGER.debug("metrics : "+ metrics);
```

Comment (FeizNouri) [reply gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java]: done

**Pull request merged**